tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/LocaleCompletion.java

```diff
@@ -149,47 +143,73 @@ public void valueChanged(String xpath, XMLSource source) {
      * @param cldrLocale the locale
      * @param stFactory the STFactory
      * @return the response
-     * @throws ExecutionException
      */
-    static LocaleCompletionResponse handleGetLocaleCompletion(final CLDRLocale cldrLocale, final STFactory stFactory) throws ExecutionException {
+    static LocaleCompletionResponse handleGetLocaleCompletion(final CLDRLocale cldrLocale, final STFactory stFactory) {
         // we need an XML Source to receive notification.
         // This causes LocaleCompletionHelper.INSTANCE.valueChanged(...) to be called
         // whenever a vote happens.
         final XMLSource mySource = stFactory.makeSource(cldrLocale.toString(), false);
         mySource.addListener(LocaleCompletion.LocaleCompletionHelper.INSTANCE);
+        return new LocaleCompletionCounter(cldrLocale, stFactory).getResponse();
+    }
 
-        LocaleCompletionResponse lcr = new LocaleCompletionCounter(cldrLocale, stFactory).getResponse();
-        // add in baseline
-        lcr.baselineCount = LocaleCompletionHelper.INSTANCE.basecache.get(cldrLocale);
-        return lcr;
+    public static int getBaselineCount(CLDRLocale cldrLocale) {
+        try {
+            return LocaleCompletionHelper.INSTANCE.basecache.get(cldrLocale);
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+            return 0;
+        }
```

Comment:
why?

Reply:
it saves adding "throws ExecutionException" to every caller -- if it's problematic, though, I'll change it

Reply:
Hm, ok i see you are calling getBaselineCount in another context.  I still think it should propagate, because the information isn't available if something goes wrongfetching